src/test/java/org/ambraproject/rhino/service/impl/ArticleListCrudServiceImplTest.java

```diff
@@ -0,0 +1,404 @@
+    );
+
+    Set<ArticleIdentifier> newArticleIds = new HashSet<>();
+    newArticleIds.add(ArticleIdentifier.create(article.getDoi()));
+
+    ArticleListView newArticleListView = mockArticleListCrudService.update(identity, Optional.empty(), Optional.ofNullable(newArticleIds));
+    assertThat(newArticleListView.getIdentity()).isEqualTo(identity);
+    assertThat(newArticleListView.getArticleList().getArticles().size()).isEqualTo(newArticleIds.size());
+  }
+
+  @Test
+  public void testUpdate_SucceedForRemoveArticlesList() {
+    ArticleListIdentity identity = createStubArticleListIdentity();
+    ArticleListView articleListView = createStubArticleList(null, null, createStubArticle());
+
+    when(mockHibernateTemplate.execute(any())).thenReturn(
+        new ArrayList(Collections.singletonList(new Object[] {
+            "valid-journal", articleListView.getArticleList()
+        })));
+
+
+    Set<String> articleKeys = new HashSet<String>();
+    articleKeys.add(createStubArticleDoi().toString());
+
+    when(mockHibernateTemplate.findByNamedParam("from Article where doi in :articleKeys",
+        "articleKeys", articleKeys)).thenReturn(
+        new ArrayList()
+    );
+
+    Set<ArticleIdentifier> newArticleIds = new HashSet<>();
+
+    ArticleListView newArticleListView = mockArticleListCrudService.update(identity, Optional.empty(), Optional.ofNullable(newArticleIds));
+    assertThat(newArticleListView.getIdentity()).isEqualTo(identity);
+    assertThat(newArticleListView.getArticleList().getArticles().size()).isEqualTo(newArticleIds.size());
+  }
+
+  @Test
+  public void testRead_FailForNonExistent() throws Exception {
+    ArticleListIdentity identity = createStubArticleListIdentity("invalid-type", "invalid-journal", "invalid-key");
+    try {
+      mockArticleListCrudService.read(identity);
+      fail("Expecting exception, but nothing was thrown.");
+    } catch (Exception exception) {
+      assertThat(exception).isInstanceOf(RestClientException.class);
+      assertThat(exception).hasMessageThat().isEqualTo("List does not exist: " + identity);
+    }
+  }
+
+  @Test
+  public void testRead_Succeed() throws Exception {
+    ArticleListIdentity identity = createStubArticleListIdentity();
+    ArticleListView articleListView = createStubArticleList();
+    ArticleList articleList = articleListView.getArticleList();
+
+    when(mockHibernateTemplate.execute(any())).thenReturn(
+        new ArrayList(Collections.singletonList(new Object[] {
+            "valid-journal", articleList
+        })));
+
+    ServiceResponse<ArticleListView> response = mockArticleListCrudService.read(identity);
+
+    ArticleListView newArticleListView = response.getBody();
+    assertThat(newArticleListView.getIdentity()).isEqualTo(identity);
+    assertThat(newArticleListView.getArticleList()).isEqualTo(articleList);
+
+//    // commented out code is needed to test without modifying ServiceResponse.java
```

Comment:
Lets resolve this - I don't like the idea of committing commented-out code. Modifying the source is fine in my opinion.

Reply:
removed